block/src/Pos/Block/Pure.hs

```diff
@@ -59,9 +60,9 @@ data VerifyHeaderParams = VerifyHeaderParams
       -- ^ Check that header has no unknown attributes.
     } deriving (Eq, Show)
 
-verifyFromEither :: Text -> Either a b -> VerificationRes
-verifyFromEither txt (Left _)  = verifyGeneric [(False, txt)]
-verifyFromEither txt (Right _) = verifyGeneric [(True, txt)]
+verifyFromEither :: Text -> Either Text () -> VerificationRes
```

Comment:
Doesn't this violate those exception guidelines? @int-index

Reply:
an ADT would be definitely preferable to raw `Text`